_posts/2017-07-06-hitchhikers-guide-to-cp.md

```diff
@@ -0,0 +1,308 @@
+---
+layout: post
+title:  "Hitchhiker's Guide to Competitive Programming"
+date:   2017-07-06
+author: udit
+---
+
+# Introduction
+
+Competitive programming, as the name suggests, is a sport of programming. One is presented with few problems for which one has to code solutions that should be efficient enough to fit in the time and memory constraints. Online judges like Codechef, Codeforces, Topcoder etc frequently organize programming competitions that one can take part in and win prizes. Few prestigious competitions like [ACM-ICPC](https://en.wikipedia.org/wiki/ACM_International_Collegiate_Programming_Contest), [Google Code Jam](https://en.wikipedia.org/wiki/Google_Code_Jam) and [Facebook Hacker Cup](https://en.wikipedia.org/wiki/Facebook_Hacker_Cup) are held annually. Doing great in these contests is an achievement in itself.
+
+Now the real question, **"Why should one try it?"**. Well, the answer is simple, programming is a major portion of the work that is done in tech companies. Being good at programming makes it easier to get a job in a good tech company. Competitive programming is the best way of getting good at programming and problem solving. Apart from this, one gets recognition and meets other programmers. There are prizes for top performers in each of these competitions. Most importantly, it quenches the thirst for those looking for challenges.
```

Comment:
Competitive programming is the quintessential Computer Science sport. It is the geek version of
your favorite sport, be it Cricket or Football or whatever. The level of competition is fierce and the passion among the participants is awesome. Also, it is a really fun way to meet and socialize with some really cool people. NITH has regular programming meetups where anyone interested can come and solve really hard problems in really short amounts of time with kindred spirits. CP is also a really good method of adding some challenges into the generally laid-back life of a CS student at NITH. 

Besides, if one wants to look at actual material advantages of competitive programming, look no further than interviews at top CS companies. To a person who has been taking part in programming contests for her entire college career, the questions asked in these interviews will generally seem to be easy mode (ok, maybe not, but the problems won't seem impossibly hard, at least).